src/clients/FtpGateway.cc

```diff
@@ -2579,7 +2579,7 @@ Ftp::Gateway::haveParsedReplyHeaders()
     // makePublic() if allowed/possible or release() otherwise
     if (flags.authenticated || // authenticated requests can't be cached
             getCurrentOffset() ||
-            !e->makePublic()) {
+            !e->makePublic(false)) {
```

Comment:
We have discussed that added `makePublic(false)` deserve analysis. I think this particular case matches the suggested "share if possible" default -- the caller does want to share this otherwise-cachable entry, even if it cannot be made public.

N.B. The effect of failed `makePublic()` is probably the same as of the `release()` call below. That existing `release()` call is OK and does not contradict `shareableWhenPrivate` "default" suggested for the `makePublic()` call above.